starboard/nplb/maximum_player_configuration_explorer.cc

```diff
@@ -0,0 +1,274 @@
+// Copyright 2023 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "starboard/nplb/maximum_player_configuration_explorer.h"
+
+#include <algorithm>
+#include <stack>
+#include <unordered_map>
+
+#include "starboard/common/log.h"
+#include "starboard/nplb/player_test_util.h"
+#include "starboard/player.h"
+
+namespace starboard {
+namespace nplb {
+
+namespace {
+
+class HashFunction {
+ public:
+  size_t operator()(const std::vector<int>& vector) const {
+    size_t hash = 0;
+    for (const auto& v : vector) {
+      hash += v;
+      hash *= 10;  // since we support at most 7 instances, 10 is good enough.
+    }
+    return hash;
+  }
+};
+
+}  // namespace
+
+bool PosetGreaterThanOrEqualTo(const std::vector<int>& a,
+                               const std::vector<int>& b) {
```

Comment:
We may also use `a >= b` here (https://en.cppreference.com/w/cpp/container/vector/operator_cmp).

Reply:
Yes, it's important to note that in this example, the operator used is for lexicographic sorting.

In this case, the ">=" operator signifies that for all indices i, a[i] must be greater than or equal to b[i]. For example, (2, 1, 1) > (1, 1, 1), but we cannot directly compare (2, 0, 1) and (1, 1, 1)  ((2, 0, 1) >= (1, 1, 1) is NOT true).

However, using the standard library's comparison (cmp) operator, both (2, 1, 1) > (1, 1, 1) and (2, 0, 1) > (1, 1, 1) would be considered true.

If there exists a standard operator that provides this functionality, it would be preferable to use it instead of creating a custom function.

Thank you!

Reply:
Yes, you are right.  I missed that details when I was reading it.